### PR TITLE
Migrate to tf2

### PR DESCRIPTION
**Pull request by f-fl0**
This PR replaces calls to `tf` functions with their `tf2` equivalent.

### CODE REVIEW

**Comment by at-wat** [comment]
## [45] FAILED on noetic

```catkin_make``` failed

**Comment by at-wat** [comment]
## [45] FAILED on kinetic

```catkin_make``` failed

**Comment by at-wat** [comment]
## [46] PASSED on melodic

<details><summary>All tests passed</summary>


```
build/test_results/ypspur_ros/roslint-ypspur_ros.xml: 1 tests
build/test_results/ypspur_ros/rostest-test_test_joint_trajectory.xml: 1 tests
build/test_results/ypspur_ros/rosunit-test_joint_trajectory.xml: 2 tests
Summary: 4 tests, 0 errors, 0 failures, 0 skipped
```
</details>

File: src/ypspur_ros.cpp [hunk 1]
```diff
@@ -1005,10 +1014,11 @@ class YpspurRosNode
         }
         pubs_["joint"].publish(joint);
 
+        tf2::Quaternion quat_tf;
         for (unsigned int i = 0; i < joints_.size(); i++)
         {
-          joint_trans[i].transform.rotation =
-              tf::createQuaternionMsgFromYaw(joint.position[i]);
+          quat_tf.setRPY(0, 0, joint.position[i]);
```
Comment (at-wat) [inline src/ypspur_ros.cpp:1020]:
```suggestion
          const tf2::Quaternion quat_tf(0, 0, joint.position[i]);
```
api ref: http://docs.ros.org/en/jade/api/tf2/html/classtf2_1_1Quaternion.html#a29cf4deb42a3a1867a066b9f74578bf6

File: src/ypspur_ros.cpp [hunk 2]
```diff
@@ -1005,10 +1014,11 @@ class YpspurRosNode
...
+          joint_trans[i].transform.rotation = tf2::toMsg(quat_tf);
```
Comment (at-wat) [inline src/ypspur_ros.cpp:1021]: or
```suggestion
          joint_trans[i].transform.rotation = tf2::toMsg(tf2::Quaternion(0, 0, joint.position[i]));
```
would be more simple.

File: src/ypspur_ros.cpp (hunk 1 above)
Comment (f-fl0) [reply src/ypspur_ros.cpp:1020]: This method seems to be deprecated http://docs.ros.org/en/noetic/api/tf2/html/classtf2_1_1Quaternion.html#a3e763411f10c67bd23bbfcc1b880793f. We can use this one though: http://docs.ros.org/en/noetic/api/tf2/html/classtf2_1_1Quaternion.html#ae23b2f32c74adb17fceb71187346f3af

Comment (f-fl0) [reply src/ypspur_ros.cpp:1020]:
```
trans.transform.rotation = tf2::toMsg(tf2::Quaternion(tf2::Vector3(0, 0, 1), msg->position[i]));
```

**Comment by at-wat** [comment]
## [47] PASSED on noetic

<details><summary>All tests passed</summary>


```
build/test_results/ypspur_ros/roslint-ypspur_ros.xml: 1 tests
build/test_results/ypspur_ros/rostest-test_test_joint_trajectory.xml: 1 tests
build/test_results/ypspur_ros/rosunit-test_joint_trajectory.xml: 2 tests
Summary: 4 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [47] PASSED on kinetic

<details><summary>All tests passed</summary>


```
build/test_results/ypspur_ros/roslint-ypspur_ros.xml: 1 tests
build/test_results/ypspur_ros/rostest-test_test_joint_trajectory.xml: 1 tests
build/test_results/ypspur_ros/rosunit-test_joint_trajectory.xml: 2 tests
Summary: 4 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [47] PASSED on melodic

<details><summary>All tests passed</summary>


```
build/test_results/ypspur_ros/roslint-ypspur_ros.xml: 1 tests
build/test_results/ypspur_ros/rostest-test_test_joint_trajectory.xml: 1 tests
build/test_results/ypspur_ros/rosunit-test_joint_trajectory.xml: 2 tests
Summary: 4 tests, 0 errors, 0 failures, 0 skipped
```
</details>

File: src/ypspur_ros.cpp (hunk 1 above)
Comment (f-fl0) [reply src/ypspur_ros.cpp]: Done in f8fbe7eca99856408d717dfb69097cc08c6fd596

File: src/ypspur_ros.cpp [hunk 3]
```diff
@@ -88,6 +88,7 @@ class YpspurRosNode
   tf2_ros::Buffer tf_buffer_;
   tf2_ros::TransformListener tf_listener_;
   tf2_ros::TransformBroadcaster tf_broadcaster_;
+  tf2::Vector3 z_axis_;
```
Comment (at-wat) [inline src/ypspur_ros.cpp:91]: Could it be const?

**Comment by at-wat** [comment]
## [48] PASSED on melodic

<details><summary>All tests passed</summary>


```
build/test_results/ypspur_ros/roslint-ypspur_ros.xml: 1 tests
build/test_results/ypspur_ros/rostest-test_test_joint_trajectory.xml: 1 tests
build/test_results/ypspur_ros/rosunit-test_joint_trajectory.xml: 2 tests
Summary: 4 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [48] PASSED on kinetic

<details><summary>All tests passed</summary>


```
build/test_results/ypspur_ros/roslint-ypspur_ros.xml: 1 tests
build/test_results/ypspur_ros/rostest-test_test_joint_trajectory.xml: 1 tests
build/test_results/ypspur_ros/rosunit-test_joint_trajectory.xml: 2 tests
Summary: 4 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Review by at-wat — APPROVED** [review]
LGTM

**Pull request merged**